### PR TITLE
Add Die Hard example from Lamport's TLA+ video course.

### PR DESCRIPTION
**Pull request by stevana**
This example passes, but I'd expect it to fail showing us one of the correct solutions to the puzzle (measuring exactly 4L using a 3 and a 5L jug), e.g. FillBig, BigIntoSmall, EmptySmall, BigIntoSmall, FillBig, BigIntoSmall.

It could be that I made a mistake somewhere, but I can't see it.

By the way, it would be nice if we could somehow "explore" the state machine by e.g. feeding it a list of labels and let it display the trace of how the states evolve...

### CODE REVIEW

File: tasty/HasCal/Test/DieHard.hs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                _big   = 0
+            return Global{..}
+
+        , coroutine = Coroutine
+            { startingLabel  = Init
+            , startingLocals = pure ()
+            , process        = init
+            }
+
+        , property = arr predicate
```
Comment (Gabriella439) [inline tasty/HasCal/Test/DieHard.hs]: I believe this is the most likely reason why the model unexpectedly passes:

```suggestion
        , property = always . arr predicate
```

HasCal differs from PlusCal in one important way, which is that it uses the `property` field for specifying both invariants and temporal properties (unlike PlusCal, which I believe specifies them separately).  That means that invariants need to be explicitly promoted to temporal properties by wrapping them in `always`.  In the absence of an `always` the predicate is only checked for the first output

Comment (Gabriella439) [reply tasty/HasCal/Test/DieHard.hs]: Oh, it looks like you just made the same change, so this is addressed 🙂

Comment (stevana) [reply tasty/HasCal/Test/DieHard.hs]: Just saw your comment in the other PR and fixed it, thanks!

**Comment by stevana** [comment]
```
- { Global: { Big: 0, Small: 0 }, Label: "Init" }
- { Global: { Big: 0, Small: 0 }, Label: "FillSmall" }
- { Global: { Big: 0, Small: 3 }, Label: "FillSmall" }
- { Global: { Big: 0, Small: 3 }, Label: "FillBig" }
- { Global: { Big: 5, Small: 3 }, Label: "FillSmall" }
- { Global: { Big: 5, Small: 3 }, Label: "FillBig" }
- { Global: { Big: 5, Small: 3 }, Label: "EmptySmall" }
- { Global: { Big: 5, Small: 0 }, Label: "FillSmall" }
- { Global: { Big: 5, Small: 3 }, Label: "EmptyBig" }
- { Global: { Big: 0, Small: 3 }, Label: "EmptySmall" }
- { Global: { Big: 0, Small: 0 }, Label: "FillBig" }
- { Global: { Big: 5, Small: 0 }, Label: "FillBig" }
- { Global: { Big: 5, Small: 0 }, Label: "EmptySmall" }
- { Global: { Big: 5, Small: 0 }, Label: "SmallToBig" }
- { Global: { Big: 5, Small: 0 }, Label: "BigToSmall" }
- { Global: { Big: 2, Small: 3 }, Label: "FillSmall" }
- { Global: { Big: 2, Small: 3 }, Label: "EmptySmall" }
- { Global: { Big: 2, Small: 0 }, Label: "FillSmall" }
- { Global: { Big: 2, Small: 3 }, Label: "EmptyBig" }
- { Global: { Big: 0, Small: 3 }, Label: "EmptyBig" }
- { Global: { Big: 0, Small: 3 }, Label: "SmallToBig" }
- { Global: { Big: 3, Small: 0 }, Label: "FillSmall" }
- { Global: { Big: 3, Small: 3 }, Label: "FillSmall" }
- { Global: { Big: 3, Small: 3 }, Label: "SmallToBig" }
- { Global: { Big: 5, Small: 1 }, Label: "FillBig" }
- { Global: { Big: 5, Small: 1 }, Label: "EmptyBig" }
- { Global: { Big: 0, Small: 1 }, Label: "EmptyBig" }
- { Global: { Big: 0, Small: 1 }, Label: "SmallToBig" }
- { Global: { Big: 1, Small: 0 }, Label: "FillSmall" }
- { Global: { Big: 1, Small: 3 }, Label: "FillSmall" }
- { Global: { Big: 1, Small: 3 }, Label: "SmallToBig" }
```

**Comment by Gabriella439** [comment]
Awesome!  Thank you for contributing this! 🙂

**Pull request merged**